### PR TITLE
Modify actions to avoid processing when only doc PRs

### PR DESCRIPTION
**Pull request by munapower**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing changes and/or breaks backward compatability?**:

```

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -53,7 +49,7 @@ jobs:
           make -C utils/docker build build-dev
 
       - name: fetch dev image
-        if: steps.filter.outputs.docker == 'false' 
+        if: steps.filter.outputs.docker == 'false'
```
Comment (mbrandenburger) [inline .github/workflows/test.yml:52]:
```suggestion
        if: steps.filter.outputs.docker == 'false'
```

**Comment by munapower** [comment]
@mbrandenburger please review PR when you can.

File: .github/workflows/documentation.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+name: Update Documentation
```
Comment (mbrandenburger) [inline .github/workflows/documentation.yml:5]:
```suggestion
name: Check Documentation
```

**Review by mbrandenburger — APPROVED** [review]
This looks good. I suggest that we focus on the documentation check (e.g., spell checking) in a follow up PR.

Thanks Maria

**Pull request merged**